File: app_project/backend/app/schemas/news/festival.py
from pydantic import BaseModel
from datetime import datetime

class FestivalBase(BaseModel):
    name: str
    description: str | None = None
    start_date: datetime
    end_date: datetime | None = None
    location: str | None = None
    is_active: bool = True

class FestivalCreate(FestivalBase):
    image: bytes | None = None

class FestivalUpdate(FestivalBase):
    image: bytes | None = None

class FestivalResponse(FestivalBase):
    id: int
    created_at: datetime
    updated_at: datetime
    image: bytes | None = None

    class Config:
        from_attributes = True

File: app_project/frontend/src/models/news/festival.ts
// models/news/festival.ts

export interface FestivalBase {
  name: string;
  description?: string;
  start_date: string; // ISO string format
  end_date?: string;
  location?: string;
  is_active: boolean;
}

export interface FestivalCreate extends FestivalBase {
  image?: File | null;
}

export interface FestivalUpdate extends FestivalBase {
  image?: File | null;
}

export interface FestivalResponse extends FestivalBase {
  id: number;
  created_at: string;
  updated_at: string;
  image?: string | null; // base64 encoded
}
